cmd/osbuild-image-tests/osbuild-image-tests.go

```diff
@@ -197,52 +519,54 @@ func getAllCases() ([]string, error) {
 }
 
 // runTests opens, parses and runs all the specified testcases
-func runTests(cases []string) error {
+func runTests(cases []string) (bool, error) {
+	testsSuccessful := true
 	for _, path := range cases {
 		f, err := os.Open(path)
 		if err != nil {
-			return fmt.Errorf("%s: cannot open test case: %v", path, err)
+			return false, fmt.Errorf("%s: cannot open test case: %v", path, err)
 		}
 
 		var testcase testcaseStruct
 		err = json.NewDecoder(f).Decode(&testcase)
 		if err != nil {
-			return fmt.Errorf("%s: cannot decode test case: %v", path, err)
+			return false, fmt.Errorf("%s: cannot decode test case: %v", path, err)
 		}
 
 		currentArch := common.CurrentArch()
-		if testcase.Compose.Arch != currentArch {
-			log.Printf("%s: skipping, the required arch is %s, the current arch is %s", path, testcase.Compose.Arch, currentArch)
+		if testcase.ComposeRequest.Arch != currentArch {
+			log.Printf("%s: skipping, the required arch is %s, the current arch is %s", path, testcase.ComposeRequest.Arch, currentArch)
 			continue
 		}
 
 		hostDistroName, err := distro.GetHostDistroName()
```

Comment:
This is unrelated to this PR, but in general I think we may want to test all the distros we can, not just the host. We want to be able to cross-build. Maybe that can be in a follow-up once we actually add support for that though.